lib/embed/media_tag.rb

```diff
@@ -149,10 +160,6 @@ def enabled_streaming_types
       Settings.streaming[:source_types]
     end
 
-    def file_index
```

Comment:
discussed in person w/ @jkeck and he was 👌 w/ this change.

(i'd refactored this because i found the original version, where the counter was read inside a method but incremented in another place by the method's calling loop, to be a bit hard to read/follow, and while the parameter passing is a bit more verbose than i'd like, it did also feel a bit cleaner and easier to follow)